AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefault.m

```diff
@@ -231,4 +236,10 @@ - (void)setMaxStorageSize:(long)sizeInBytes completionHandler:(nullable void (^)
   });
 }
 
+- (void)authTokenContext:(__unused MSAuthTokenContext *)authTokenContext didReceiveAuthToken:(NSString *)authToken {
+  if (self.ingestion) {
```

Comment:
I think I wouldn't assign the authToken to ingestion. It's a tied to a batch of logs but which batch of logs needs what token is a channel unit logic. The current design would work at the moment but I am concerned about the near future.

Reply:
But we give all other headers to ingestion and channels do not know about headers. What problem do you foresee? And the solution for making sure the logs get the right headers will still be managed by channel since it decides when to flush, right?

I also think that we can do that change if we need to, but not necessary to do that now. Doing it later wouldn't be a big change, and we may not even need to.

Reply:
(I think the order could be flush, and then once ingestion makes the request we update its auth token)

Reply:
OK let's go with this for now, keeping it simple and we'll adjust later as needed.

Reply:
I agree with you both guys but we need to have a discussion about it before moving forward. This is like a design concern so we should have agreement and be on the same page.